src-java/reroute-topology/reroute-storm-topology/src/main/java/org/openkilda/wfm/topology/reroute/service/RerouteService.java

```diff
@@ -102,7 +102,12 @@ public void rerouteAffectedFlows(MessageSender sender, String correlationId, Rer
             Flow flow = entry.getFlow();
             Boolean sendRerouteRequest = transactionManager.doInTransaction(() -> {
                 boolean flowPathFound = updateFlowPathsStateForFlow(switchId, port, entry.getAffectedPaths());
-                flowRepository.updateStatusSafe(flow.getFlowId(), flow.computeFlowStatus());
+                FlowStatus flowStatus = flow.computeFlowStatus();
+                String flowStatusInfo = null;
+                if (!FlowStatus.UP.equals(flowStatus)) {
```

Comment:
How it is possible? If the flow is the "target" for rerouting, how it can be in UP state?

Reply:
Before that, we update the paths statuses and compute the flow status.